contracts/TokenFactory.sol

```diff
@@ -326,18 +358,71 @@ contract TokenFactory is ReentrancyGuard, Ownable {
         }
     }
 
-    function burnTokenAndMintWinner(
+    function publishToUniswap (
         address tokenAddress
     ) external nonReentrant {
         uint256 _competitionId = competitionIds[tokenAddress];
+
+        require(_competitionId != currentCompetitionId, 'The competition is still active');
+
+        address winnerToken = getWinnerByCompetitionId(_competitionId);
+
+        require(winnerToken == tokenAddress, 'Token address not winner');
+
+        uint256 totalCollateral = getCollateralByCompetitionId(_competitionId);
+        WETH.deposit{value: totalCollateral}();
+
+        // calulate winner token amount
+        uint256 contributionWithoutFee = (totalCollateral * FEE_DENOMINATOR) /
+            (FEE_DENOMINATOR + feePercent);
+
+        uint256 winnerTokenAmount = bondingCurve.computeMintingAmountFromPrice(
+            totalCollateral,
+            Token(tokenAddress).totalSupply(),
+            contributionWithoutFee
```

Comment:
What we want to know here is the per-token (1e18) unit price of the token, so this parameter should be 1.0 (1e18). After getting the per-unit price, we can then calculate the amount of tokens to be minted by totalCollateral / unitPrice (in decimal space), which would give us the number of units, i.e. unwrap(div(ud(totalCollateral), ud(unitPrice)))